pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1395,6 +1395,14 @@ func (c *PodAffinityChecker) satisfiesPodsAffinityAntiAffinity(pod *v1.Pod, node
 
 // PodToleratesNodeTaints checks if a pod tolerations can tolerate the node taints
 func PodToleratesNodeTaints(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	if nodeInfo == nil || nodeInfo.Node() == nil {
+		return false, []algorithm.PredicateFailureReason{ErrNodeUnknownCondition}, nil
+	}
+
+	if nodeInfo.Node().Spec.Unschedulable {
```

Comment:
This function, as its name suggests, should only check if the pod tolerates node taints. Because this is also used by DaemonSet controller, it causes non-backward compatible changes to DaemonSet, see #60163.

Reply:
+1, let me handle it :)

Reply:
Thanks @k82cn!